lexicon/providers/rcodezero.py

```diff
@@ -0,0 +1,196 @@
+                        }
+                        records.append(processed_record)
+
+        LOGGER.debug('list_records: %s', records)
+        return records
+
+    def _update_record(self, identifier, rtype=None, name=None, content=None):
+        self._delete_record(identifier, rtype, name, None)
+        return self._create_record(rtype, name, content)
+
+    # Delete an existing record.
+    # If record does not exist, do nothing.
+    def _delete_record(self, identifier=None, rtype=None, name=None, content=None):
+
+        if identifier is not None:
+            rtype, name, content = self._parse_identifier(identifier)
+
+        LOGGER.debug("delete %s %s %s", rtype, name, content)
+        if rtype is None or name is None:
+            raise Exception("Must specify at least both rtype and name")
+
+        payload = self._get(
+            '/zones/{0}/rrsets?page_size=-1'.format(self.domain_id))
+
+        for rrset in payload['data']:
+            if rrset['type'] == rtype and self._fqdn_name(rrset['name']) == self._fqdn_name(name):
+                update_data = rrset
+
+                if content is None:
+                    update_data['records'] = []
+                    update_data['changetype'] = 'DELETE'
+                else:
+                    new_record_list = []
+                    for record in update_data['records']:
+                        if self._clean_content(rrset['type'], content) != record['content']:
+                            new_record_list.append(record)
+
+                    update_data['records'] = new_record_list
+                    update_data['changetype'] = 'UPDATE'
+                break
+
+        request = [update_data]
+        LOGGER.debug('request: %s', request)
+
+        self._patch('/zones/' + self.domain + '/rrsets', data=request)
+        return True
+
+    # Helpers
+    def _request(self, action='GET', url='/', data=None, query_params=None):
+        if data is None:
+            data = {}
+        if query_params is None:
+            query_params = {}
+        response = requests.request(action, self.api_endpoint + url, params=query_params,
+                                    data=json.dumps(data),
+                                    headers={
+                                        'Authorization': 'Bearer ' +
+                                                         self._get_provider_option('auth_token'),
+                                        'Content-Type': 'application/json'
+                                    })
+        # if the request fails for any reason, throw an error.
+        if response.status_code >= 400:
+            LOGGER.error('Bad Request: %s', response.text)
+        response.raise_for_status()
+        return response.json()
+
+    # generate a unique id for a give record
+    def _make_identifier(self, rtype, name, content):  # pylint: disable=no-self-use
+        return "{}/{}={}".format(rtype, name, content)
```

Comment:
For the sake of CLI usability, could you hash this identifier to have a short string exposed to the user as an ID? An example code can be found here: https://github.com/AnalogJ/lexicon/blob/cc985166aead697727e56517c58061693ea087ce/lexicon/providers/godaddy.py#L227-L233

Reply:
fixed in e865177